text/0068-entity-tags.md

```diff
@@ -0,0 +1,309 @@
+
+Here's a policy that conforms to this schema:
+```
+permit (
+  principal is User,
+  action == Action::"writeDoc",
+  resource is Document) 
+when {
+  document.owner == principal ||
+    (principal.jobLevel > 6 &&
+    resource.policyTags has "write" && 
+    principal.authTags has "write" && 
+    resource.policyTags["write"].containsAny(principal.authTags["write"]))
+};
+```
+This policy states that for a `User` to carry out the _writeDoc_ action on a `Document`, the user must own the document, or else the user's job level must be at least 6 and the document and the user must each have a `write` tag in their attached tags, where at least one of the user's write-tag's values must be present in the document's write-tag's values.
+
+## Detailed design
+
+Neither the Cedar evaluator/authorizer, the JSON format for entities, nor the JSON or natural syntax for policies needs to change to support entity tags. This is because tags are represented as records and support the same operations. Only the Cedar schema format, the validator, and the schema-based entity parser need to change to account for a new `{ T }` type.
+
+### Schema
+
+Tags have type ` { T }`, where `T` is the type of tag values. Only entity attributes can be given type `{ T }`, and `T` cannot directly mention another tags type.
+
+We extend schemas as follows to support the new tags type. Here's the natural syntax:
+```
+Entity           := 'entity' Idents ['in' EntOrTyps] [['='] EntityRecType] ';'
+EntityRecType    := '{' [EntityAttrDecls] '}'
+EntityAttrDecls  := Name ['?'] ':' [Type | TagsType] [',' | ',' EntityAttrDecls]
+TagsType         := '{' Type '}'
+Type             := PRIMTYPE | Path | SetType | RecType
+```
+In essence: We alter the definition of entity types to include attributes with type `{ T }`, where `T` is any type not mentioning another tags type. Not shown here are the productions for `RecType` and `AttrDecls`, which apply to records rather than entities. These productions are unchanged---normal records may not include attributes with `{ T }` types.
+
+The JSON syntax for schemas specifies tags as records with a `default` type. Doing so leverages the analogy that tags are like records in which all attributes are optional, with the same type. Here's our introductory example schema in this format:
+```
+"entityTypes": {
+    "User" : {
+        "shape" : {
+            "type" : "Record",
+            "attributes" : {
+                "jobLevel" : {
+                    "type" : "Long"
+                },
+                "authTags" : {
+                    "type" : "Record",
+                    "attributes" : {
+                        "default": {
+                            "type" : "Set",
+                            "element": "String"
+                        }
```

Comment:
What does the JSON for 
```
authTags: { default: Set<String> },
```
look like, and how does it compare to the JSON for
```
authTags: { Set<String> },
```
?

(Unless I'm missing something -- which is very possible -- both of these are valid in human-syntax schemas, assuming that we're inside an entity and not a record, because `default` is not a reserved attribute name; I assume this RFC isn't proposing that we reserve the attribute name `default`, which would potentially be disruptive.)

Reply:
Good catch! I've updated with the most recent commit.

Reply:
Fix looks great